Gemfile

```diff
@@ -15,6 +15,8 @@ gem "rails-i18n"
 gem "sprockets-rails", "~> 3.2", ">= 3.2.1"
 gem "jquery-rails", "~> 4.4"
 gem "bootstrap", "~> 4.5"
+gem "bcrypt", "~> 3.1", ">= 3.1.13"
+gem "config", "~> 2.2", ">= 2.2.1"
```

Comment:
gem này có tác dụng gì

Reply:
+ bcrypt cung cấp các thuật toán để xử lí mã hoá và giải mã mật khẩu (sử dụng hàm băm) an toàn... ngoài ra nó cũng cung cấp sẵn các cách xử lí tương ứng cho việc implement các thuật toán đó trong Rails
+ config thì cung cấp cho mình các cách để tạo ra các settings đa môi trường cho các dự án Rails